lib/cretonne/src/isa/intel/abi.rs

```diff
@@ -168,6 +169,31 @@ pub fn callee_saved_registers(flags: &shared_settings::Flags) -> &'static [RU] {
     }
 }
 
+fn callee_saved_registers_used(
+    flags: &shared_settings::Flags,
+    func: &ir::Function,
+) -> AllocatableSet {
+    let mut all_callee_saved = AllocatableSet::empty();
+    for reg in callee_saved_registers(flags) {
+        all_callee_saved.free(GPR, *reg as RegUnit);
+    }
+
+    let mut used = AllocatableSet::empty();
+    for value_loc in func.locations.values() {
+        match value_loc {
+            &ValueLoc::Reg(ru) => {
+                if !used.is_avail(GPR, ru) {
+                    used.free(GPR, ru);
```

Comment:
A subtlety here is that the ValueLoc here only contains one register unit of a potentially multi-unit register. This doesn't occur on x86, because we've decided to ignore the H registers in the register allocator. But it wouldn't necessarily be ok for other architectures, so we should put comments here advising people not to blindly reuse this code without thinking about it.